src/main/java/org/opensearch/neuralsearch/processor/InferenceProcessor.java

```diff
@@ -154,9 +164,20 @@ Map<String, Object> buildMapWithProcessorKeyAndOriginalValue(IngestDocument inge
         for (Map.Entry<String, Object> fieldMapEntry : fieldMap.entrySet()) {
             String originalKey = fieldMapEntry.getKey();
             Object targetKey = fieldMapEntry.getValue();
+
+            int nestedDotIndex = originalKey.indexOf('.');
+            if (nestedDotIndex != -1) {
+                Map<String, Object> newTargetKey = new LinkedHashMap<>();
+                newTargetKey.put(originalKey.substring(nestedDotIndex + 1), targetKey);
+                targetKey = newTargetKey;
+
+                originalKey = originalKey.substring(0, nestedDotIndex);
+            }
```

Comment:
@Sanjana679 can you please provide details how we are handling multiple level of nesting here?

Reply:
At the moment, I'm not currently handling multiple levels of nesting, as I initially thought it was only for one level. However, I will work on handling multiple levels of nesting.